src/Datagrid/DatagridMapper.php

```diff
@@ -78,7 +81,7 @@ public function add(
     ): self {
         if ($name instanceof FieldDescriptionInterface) {
             $fieldDescription = $name;
-            $fieldDescription->mergeOptions($filterOptions);
+            $fieldDescription->mergeOptions($fieldDescriptionOptions);
```

Comment:
Not sure about this change, but it makes more sens to pass a variable called `fieldDescriptionOptions` rather than a `filterOption`. Maybe the real question is what's the differences between these two variables ?

Reply:
I also have an issue with the code 
```
    final public function mergeOptions(array $options = []): void
    {
        $this->setOptions(array_merge_recursive($this->options, $options));
    }
```
I don't think `array_merge_recursive` is doing what we're looking for ; that's why psalm emit an issue.